### ISSUE TITLE
helpful--docstring-advice failure against GNU Emacs 28.1

### ISSUE THREAD

**Issue by dogsleg** [Wilfred/helpful#302]
Hi,

Thanks for your work on helpful!

I'm getting the test failure (helpful version is 0.19) when testing against GNU Emacs 28.1 (installed from the Debian archive, package version is 1:28.1+1-4). The relevant part of the test log is as follows:

```
$ LC_ALL=C.UTF-8 emacs -batch -Q -l package --eval "(add-to-list 'package-directory-list \"/usr/share/emacs/site-lisp/elpa\")" --eval "(add-to-list 'package-directory-list \"/usr/share/emacs/site-lisp/elpa-src\")" -f package-initialize -L . -L test --eval "(load-file \"test/test-helper.el\")" -l test/helpful-unit-test.el --eval \(ert-run-tests-batch-and-exit\)

[...]

passed  22/90  helpful--docstring (0.000097 sec)
Test helpful--docstring-advice backtrace:
  signal(ert-test-failed (((should (equal (helpful--docstring #'test-f
  ert-fail(((should (equal (helpful--docstring #'test-foo-advised t) "
  (if (unwind-protect (setq value-47 (apply fn-45 args-46)) (setq form
  (let (form-description-49) (if (unwind-protect (setq value-47 (apply
  (let ((value-47 'ert-form-evaluation-aborted-48)) (let (form-descrip
  (let* ((fn-45 #'equal) (args-46 (condition-case err (let ((signal-ho
  (let ((lexical-binding t)) (let* ((fn-45 #'equal) (args-46 (conditio
  (closure (t) nil (let ((lexical-binding t)) (let* ((fn-45 #'equal) (
  ert--run-test-internal(#s(ert--test-execution-info :test #s(ert-test
  ert-run-test(#s(ert-test :name helpful--docstring-advice :documentat
  ert-run-or-rerun-test(#s(ert--stats :selector t :tests ... :test-map
  ert-run-tests(t #f(compiled-function (event-type &rest event-args) #
  ert-run-tests-batch(nil)
  ert-run-tests-batch-and-exit()
  command-line-1(("-l" "package" "--eval" "(add-to-list 'package-direc
  command-line()
  normal-top-level()
Test helpful--docstring-advice condition:
    (ert-test-failed
     ((should
       (equal
	(helpful--docstring ... t)
	"Docstring here too."))
      :form
      (equal "Docstring here too.\n\nThis function has :around advice: `ad-Advice-test-foo-advised'." "Docstring here too.")
      :value nil :explanation
      (arrays-of-different-length 84 19 "Docstring here too.\n\nThis function has :around advice: `ad-Advice-test-foo-advised'." "Docstring here too." first-mismatch-at 19)))
   FAILED  23/90  helpful--docstring-advice (0.000269 sec)

[...]

Ran 90 tests, 86 results as expected, 1 unexpected, 3 skipped (2022-09-28 17:29:57+0500, 8.382789 sec)

1 unexpected results:
   FAILED  helpful--docstring-advice

3 skipped results:
  SKIPPED  helpful--definition-c-vars
  SKIPPED  helpful--definition-special-form
  SKIPPED  helpful-callable--with-C-source
```

**Comment by AdrianBunk** [comment]
This was fixed in #296

**Comment by Wilfred** [comment]
Closing per above, feel free to reopen if I've missed anything.
